File: 2016f/soln/Greeting.java
import java.util.*;
import java.io.*;

class Greeting {
	public static void main(String[] args) throws FileNotFoundException {
		Scanner sc = new Scanner(new FileReader("GreetingIN.txt"));
		while (sc.hasNext()) {
			String name = sc.nextLine();
			greet(name);
		}
	}

	public static void greet(String name) {
		System.out.println("Hello " + name + ".");
	}
}

File: 2016f/soln/Greeting.py
def greeting(S):
    """

    :param S: A person's name
    :return: "Hello <name>"
    """

    return 'Hello {}.'.format(S)


def parse_file_and_call_function():
    with open("GreetingIN.txt", "r") as f:
        for line in f:
            print greeting(line.strip())


if __name__ == "__main__":
    parse_file_and_call_function()
